conkit/io/mapalign.py

```diff
@@ -0,0 +1,118 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2016-18, University of Liverpool
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""
+Parser module specific to map_align map files
+"""
+
+from conkit.io._parser import ContactFileParser
+from conkit.core.contact import Contact
+from conkit.core.contactmap import ContactMap
+from conkit.core.contactfile import ContactFile
+
+
+class MapAlignParser(ContactFileParser):
+    """Class to parse a map_align map file
+    """
+
+    def read(self, f_handle, f_id="map_align"):
+        """Read a contact file
+
+        Parameters
+        ----------
+        f_handle
+           Open file handle [read permissions]
+        f_id : str, optional
+           Unique contact file identifier
+
+        Returns
+        -------
+        :obj:`~conkit.core.contactfile.ContactFile`
+
+        Raises
+        ------
+        :exc:`RuntimeError`
+           Sequence length exceeds the length stated with the LEN card
+
+        """
+
+        hierarchy = ContactFile(f_id)
+        _map = ContactMap("map_1")
+        hierarchy.add(_map)
+        seq_length = None
+
+        for line in f_handle:
+            line = line.strip().split()
+
+            if line[0] == "LEN":
+                seq_length = int(line[1])
+                continue
+
+            elif line[0] != "CON":
+                continue
+
+            elif line[1].isdigit() and line[2].isdigit():
+                line.pop(0)
```

Comment:
this is weird. you remove an item when you're just indexing below. why not just 

1/ unpack the values
```
_, c1, c2, d, *_ = line
if seq_length is not None and int(c1) > seq_length or int(c2) > seq_length:
    ... 
```
2/ index the stuff below without the removal of the item